File: lib/models/transferencia.dart
import 'contato.dart';

class Transferencia {
  final double valor;
  final Contato contato;
  final DateTime data;

  Transferencia(
    this.valor,
    this.contato, {
    this.data,
  });

  @override
  String toString() {
    return 'Transferencia{valor: $valor, contato:  $contato, data: $data}';
  }

  Transferencia.deJson(Map<String, dynamic> json)
      : valor = json['valor'],
        contato = Contato.deJson(json['contato']),
        data = DateTime.parse(json['data']);

  Map<String, dynamic> paraJson() => {
        "valor": valor,
        "contato": contato.paraJson(),
      };
}

File: src/main/kotlin/br/com/alura/bytebank/model/Transferencia.kt
package br.com.alura.bytebank.model

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.*
import javax.persistence.Embedded
import javax.persistence.Entity
import javax.persistence.Id

@Entity
class Transferencia(
        @Id
        val id: String = UUID.randomUUID().toString(),
        val valor: BigDecimal,
        @Embedded
        val contato: Contato,
        val data: LocalDateTime = LocalDateTime.now()
)
